instance/main.tf

```diff
@@ -53,7 +66,7 @@ resource "aws_instance" "web_public" {
 
 
   tags = {
-    Name = "${var.env_code}-web-public"
+    Name = "${data.terraform_remote_state.networking.outputs.env_code}-web-public"
```

Comment:
Please don't use a remote state for this variable. Instead, declare it as a separate variable in level2.
The name can remain the same: level2